feature/calculator/mvp1

j-kachalova opened the pull request.

j-kachalova committed:
add dto

j-kachalova committed:
add LoanService

j-kachalova committed:
add validation for prescoring

j-kachalova committed:
add scoring (without denial)

j-kachalova committed:
add calculations for ScoringController, tests, swagger

j-kachalova committed:
add tests, change getPaymentSchedule

j-kachalova committed:
delete some logs

Skillshut commented on calculator/.mvn/wrapper/maven-wrapper.properties:
```diff
@@ -0,0 +1,18 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
нужно убрать файл из индекста гита

Skillshut commented on calculator/mvnw:
```diff
@@ -0,0 +1,250 @@
+#!/bin/sh
```
и этот

Skillshut commented on calculator/mvnw.cmd:
```diff
@@ -0,0 +1,146 @@
+<# : batch portion
```
и этот

Skillshut commented on calculator/pom.xml:
```diff
@@ -0,0 +1,97 @@
+	<version>0.0.1-SNAPSHOT</version>
+	<name>calculator</name>
+	<description>calculator</description>
+	<properties>
```
все версии зависимостей нужно вынести в  блок properties и уже ссылаться на них при использовании

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/CalculatorApplication.java:
```diff
@@ -0,0 +1,20 @@
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+import java.math.BigDecimal;
```
не нужный импорт нужно удалить. вообще для удобства автоматическое удаление неиспользуемых импортов Ctr + Alt + O, автоматическое приведение кода в соответствие code style -- Ctr + Alt + L

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/LoanOfferController.java:
```diff
@@ -0,0 +1,36 @@
+@RestController
+public class LoanOfferController {
+    @Value("${app.keyRate}")
+    BigDecimal keyRate;
```
поля должны по максимуму быть private, здесь тоже

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/LoanOfferController.java:
```diff
@@ -0,0 +1,36 @@
+    BigDecimal keyRate;
+    @Autowired
+    private LoanService loanService;
+    static Logger log = Logger.getLogger(LoanOfferController.class.getName());
```
логгер использовать нужно из lombok через аннотацию, а не создавать каждый раз его в ручную

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/LoanOfferController.java:
```diff
@@ -0,0 +1,36 @@
+public class LoanOfferController {
+    @Value("${app.keyRate}")
+    BigDecimal keyRate;
+    @Autowired
```
лучше использовать не такой способ, а тоже использовать lombok и аннотацию @ RequiredArgsConstructor. а само поле сервиса сделать final

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/LoanOfferController.java:
```diff
@@ -0,0 +1,36 @@
+
+    @PostMapping("/calculator/offers")
+    public List<LoanOfferDto> calculateLoanOffers( @RequestBody LoanStatementRequestDto request) throws ScoringdataDtoValidationExc {
+        log.info("Got LoanController" + request.toString());
```
лучше написать что-то типа LoanController request:
и не использовать toString(), он сам автоматически вызовется

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/LoanOfferController.java:
```diff
@@ -0,0 +1,36 @@
+    public List<LoanOfferDto> calculateLoanOffers( @RequestBody LoanStatementRequestDto request) throws ScoringdataDtoValidationExc {
+        log.info("Got LoanController" + request.toString());
+        List<LoanOfferDto> loanOfferDtoList = loanService.generateLoanOffers(request,keyRate);
+        log.info("Sended List<LoanOfferDto>" + loanOfferDtoList.toString());
```
LoanController response

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+import java.util.logging.Logger;
+
+@RestController
+public class ScoringController {
```
тоже самое как и контроллере другом -- @ RequiredArgsConstructor. а сами поля сервисов сделать final

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+@RestController
+public class ScoringController {
+
+    @Autowired
```
@Autowired аннотации убрать

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+    private ScoringService scoringService;
+    @Autowired
+    private CreditPaymentsService creditPaymentsService;
+    static Logger log = Logger.getLogger(ScoringController.class.getName());
```
также по логгеру как в контроллере другом написал

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+    private CreditPaymentsService creditPaymentsService;
+    static Logger log = Logger.getLogger(ScoringController.class.getName());
+    @Value("${app.keyRate}")
+    BigDecimal keyRate;
```
приватное поле нужно

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+    @ApiOperation(value = "get creditDTO", response = CreditDto.class)
+    @PostMapping("/calculator/calc")
+    public CreditDto getCredit(@RequestBody ScoringDataDto scoringDataDto) throws ScoringdataDtoValidationExc {
+        log.info("Got ScoringDataDto" +scoringDataDto.toString());
```
логирование также везде поправить

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+    public CreditDto getCredit(@RequestBody ScoringDataDto scoringDataDto) throws ScoringdataDtoValidationExc {
+        log.info("Got ScoringDataDto" +scoringDataDto.toString());
+        BigDecimal rate = scoringService.makeScoring(scoringDataDto, keyRate);
+        BigDecimal monthlyPayment = creditPaymentsService.calculateMonthlyPayment(rate,scoringDataDto.getAmount(),scoringDataDto.getTerm());
```
опечатка

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+        log.info("Got ScoringDataDto" +scoringDataDto.toString());
+        BigDecimal rate = scoringService.makeScoring(scoringDataDto, keyRate);
+        BigDecimal monthlyPayment = creditPaymentsService.calculateMonthlyPayment(rate,scoringDataDto.getAmount(),scoringDataDto.getTerm());
+        BigDecimal psk = creditPaymentsService.calculatePsk(monthlyPayment, scoringDataDto.getTerm());
```
лучше более смысловые и длинные названия давать, чтобы потом другому разработчику было проще читать код

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,43 @@
+        BigDecimal monthlyPayment = creditPaymentsService.calculateMonthlyPayment(rate,scoringDataDto.getAmount(),scoringDataDto.getTerm());
+        BigDecimal psk = creditPaymentsService.calculatePsk(monthlyPayment, scoringDataDto.getTerm());
+        List<PaymentScheduleElementDto> paymentScheduleElementDtoList = creditPaymentsService.getPaymentSchedule(scoringDataDto,monthlyPayment, rate, LocalDate.now());
+        CreditDto creditDto = new CreditDto(scoringDataDto.getAmount(), scoringDataDto.getTerm(), monthlyPayment, rate, psk, scoringDataDto.getIsInsuranceEnabled(),scoringDataDto.getIsSalaryClient(), paymentScheduleElementDtoList);
```
в место ручного создания через конструктор тоже лучше использовать Lombok и паттерн builder, соответствующая аннотация ставиться над классов dto

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/CreditDto.java:
```diff
@@ -0,0 +1,23 @@
+import java.math.BigDecimal;
+import java.util.List;
+
+@Getter
```
вот тут добавить @ builder, а также можно заменить некоторые аннотации ниже, одной -- @ data, она включает в себя несколько аннотаций

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/EmploymentDto.java:
```diff
@@ -0,0 +1,22 @@
+
+import java.math.BigDecimal;
+
+@Getter
```
аналогично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/LoanOfferDto.java:
```diff
@@ -0,0 +1,23 @@
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
```
аналогично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/LoanStatementRequestDto.java:
```diff
@@ -0,0 +1,55 @@
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
```
аналогично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/PaymentScheduleElementDto.java:
```diff
@@ -0,0 +1,20 @@
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
```
аналогично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/ScoringDataDto.java:
```diff
@@ -0,0 +1,35 @@
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
```
аналогично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/exeptions/ScoringdataDtoValidationExc.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.BAD_REQUEST)
+public class ScoringdataDtoValidationExc extends Exception {
```
нужно наследоваться от RuntimeException

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/interfaces/ScoringDataInt.java:
```diff
@@ -0,0 +1,11 @@
+import java.math.BigDecimal;
+
+public interface ScoringDataInt {
+    BigDecimal makeScoring(@Valid ScoringDataDto scoringDataDto, BigDecimal rate) throws ScoringdataDtoValidationExc;
```
и так как ScoringdataDtoValidationExc унаследовано от RuntimeException, то секцию throws нужно будет убрать

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+
+@Service
+public class CreditPaymentsService implements CreditPaymentsInt {
+    static Logger log = Logger.getLogger(CreditPaymentsService.class.getName());
```
также логгер через ломбок

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+    @Override
+    public BigDecimal calculateMonthlyPayment(BigDecimal rate, BigDecimal amount, Integer term) {
+
+        BigDecimal monthlyRate = rate.divide(BigDecimal.valueOf(1200),7, RoundingMode. HALF_UP);
```
числа все нужно выносить в константы классы или общие какие-то

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+    public BigDecimal calculateMonthlyPayment(BigDecimal rate, BigDecimal amount, Integer term) {
+
+        BigDecimal monthlyRate = rate.divide(BigDecimal.valueOf(1200),7, RoundingMode. HALF_UP);
+        BigDecimal x = monthlyRate.add(BigDecimal.valueOf(1)).pow(term);
```
числа в константы

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+
+        BigDecimal monthlyRate = rate.divide(BigDecimal.valueOf(1200),7, RoundingMode. HALF_UP);
+        BigDecimal x = monthlyRate.add(BigDecimal.valueOf(1)).pow(term);
+        BigDecimal annuityRate = monthlyRate.multiply(x).divide(x.subtract(BigDecimal.valueOf(1)), 7, RoundingMode.HALF_UP);
```
вообщем везде числа выносим в консанты, иначе через пол месяца забывается что это число значит

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+        BigDecimal x = monthlyRate.add(BigDecimal.valueOf(1)).pow(term);
+        BigDecimal annuityRate = monthlyRate.multiply(x).divide(x.subtract(BigDecimal.valueOf(1)), 7, RoundingMode.HALF_UP);
+        BigDecimal monthlyPayment = amount.multiply(annuityRate);
+        log.info("Monthly payment : " + monthlyPayment.setScale(2, RoundingMode.HALF_UP));
```
и логирование не делаем конкатенацией строк, в логере специально есть плейсхолдеры { }, и дальше в метод как параметры передаем данные которые нужно залогировать

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+    @Override
+    public BigDecimal calculatePsk(BigDecimal monthlyPayment, Integer term){
+        BigDecimal psk = monthlyPayment.multiply(BigDecimal.valueOf(term));
+        log.info("psk calculation" +psk.toString());
```
аналогично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+        BigDecimal remainingDebt=scoringDataDto.getAmount();
+        LocalDate date1 = date.plusMonths(1);
+
+        for(int i = 1;remainingDebt.compareTo(BigDecimal.ZERO)>0;i++){
```
а почему здесь не while? если условие такое.  for нагляднее для итерации по какому-то списку от начала до конца, здесь лучше while по смыслу подходит, но в принципе не критично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+            Integer number=i;
+            LocalDate date2 = date1.minusMonths(1);
+            long daysDifference = date2.until(date1, ChronoUnit.DAYS);
+            BigDecimal debtPayment = remainingDebt.multiply(rate).multiply(BigDecimal.valueOf(daysDifference*0.01)).divide(BigDecimal.valueOf(date1.lengthOfYear()), 2, RoundingMode.HALF_UP);
```
такие цепочки вызовов лучше писать не в строку, а каждый вызов метода на новой строке, это удобнее читать

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/CreditPaymentsService.java:
```diff
@@ -0,0 +1,61 @@
+            BigDecimal debtPayment = remainingDebt.multiply(rate).multiply(BigDecimal.valueOf(daysDifference*0.01)).divide(BigDecimal.valueOf(date1.lengthOfYear()), 2, RoundingMode.HALF_UP);
+            BigDecimal interestPayment = monthlyPayment.subtract(debtPayment);
+            remainingDebt=remainingDebt.subtract(interestPayment);
+            if(remainingDebt.compareTo(BigDecimal.ZERO)<0) remainingDebt = BigDecimal.ZERO;
```
на 2 строках нужно, удобнее будет читать и лучше делать явный блок со скобками, хоть тут и 1 строка, чтобы было действительно понятно, что нужно только 1 эта строка

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/LoanService.java:
```diff
@@ -0,0 +1,57 @@
+public class LoanService implements LoanOffersCounterInt {
+
+
+    private static final Logger log = LoggerFactory.getLogger(LoanService.class);
```
ломбок

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/LoanService.java:
```diff
@@ -0,0 +1,57 @@
+                createLoanOfferDto(true, true, request, baseInterestRate)
+
+        );
+        log.info("Generated loan offers list: {}", loanOfferDtoList);
```
вот тут как раз ты использовала плейсхолдер, везде надо так

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/LoanService.java:
```diff
@@ -0,0 +1,57 @@
+                                           Boolean isSalaryClient,
+                                           LoanStatementRequestDto requestDto,
+                                           BigDecimal baseInterestRate) {
+        ScoringService scoringService = new ScoringService();
```
создание объектов через контруктор нужно убрать, и сделать полем, и внедрять через спринг

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/LoanService.java:
```diff
@@ -0,0 +1,57 @@
+                                           LoanStatementRequestDto requestDto,
+                                           BigDecimal baseInterestRate) {
+        ScoringService scoringService = new ScoringService();
+        CreditPaymentsService creditPaymentsService = new CreditPaymentsService();
```
здесь аналогично

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/LoanService.java:
```diff
@@ -0,0 +1,57 @@
+        BigDecimal monthlyPayment = creditPaymentsService.calculateMonthlyPayment(rate, requestDto.getAmount(), requestDto.getTerm());
+        BigDecimal totalAmount = creditPaymentsService.calculatePsk(monthlyPayment,requestDto.getTerm());
+
+        return new LoanOfferDto(UUID.randomUUID(),
```
через builder

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+
+@Service
+@Validated
+public class ScoringService implements ScoringDataInt {
```
для всех сервисов интерфейсы лучше назвать по типу ScoringService и они (интерфейсы) лежат в папке service, а импелемнтацию сервиса назвать по типу ScoringServiceImpl и она лежит в папке service.impl

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+@Validated
+public class ScoringService implements ScoringDataInt {
+
+    static Logger log = Logger.getLogger(ScoringService.class.getName());
```
lombok

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+
+    static Logger log = Logger.getLogger(ScoringService.class.getName());
+
+    private final BigDecimal selfEmployedRateIncrease = BigDecimal.valueOf(1.0);
```
вот таки нужно делать константы, но еще добавить static модификатор и название по должно быть типа ETO_MOYA_CONSTANTA

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+    private final BigDecimal nonBinaryRateIncrease = BigDecimal.valueOf(7.0);
+    @Override
+    public BigDecimal makeScoring(@Valid ScoringDataDto scoringDataDto, BigDecimal rate) throws ScoringdataDtoValidationExc {
+        log.info("got "+scoringDataDto.toString()+rate.toString());
```
лорировать нужно по типу : request with scoringDataDto = {}, rate = {}

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+        LocalDate birthday = scoringDataDto.getBirthdate();
+        Integer age = birthday.until(LocalDate.now()).getYears();
+        log.info("birthday is "+birthday);
+        log.info("age is "+age);
```
это в один лог можно и лучше это для debug уровня, info логируем только входные и выходные параметры, внутри метода, все остальные логи для debug мода, потому что на продакшн среде логирование дополнительно будет нагружать микросервис

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+        log.info("birthday is "+birthday);
+        log.info("age is "+age);
+        rate = checkEmploymentStatus(scoringDataDto.getEmployment().getEmploymentStatus(), rate);
+        log.info("rate after checkEmploymentStatus is "+rate);
```
debug

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+        rate = checkEmploymentStatus(scoringDataDto.getEmployment().getEmploymentStatus(), rate);
+        log.info("rate after checkEmploymentStatus is "+rate);
+        rate = checkPosition(scoringDataDto.getEmployment().getPosition(), rate);
+        log.info("rate after checkPosition is "+rate);
```
и так далее

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+        return rate;
+    }
+    private void checkAmount(BigDecimal amount, BigDecimal salary) throws ScoringdataDtoValidationExc {
+        if (amount.compareTo(salary.multiply(BigDecimal.valueOf(25))) > 1) {
```
константа

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+            case TOP_MANAGER:
+                rate=rate.subtract(topManagerRateDecrease);
+                break;
+            default: break;
```
default не обязательно

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+            case DIVORCED:
+                rate = rate.add(divorcedRateIncrease);
+                break;
+            default:
```
default не обязательно

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+                    break;
+            case NON_BINARY:
+                rate=rate.add(nonBinaryRateIncrease);
+            default:break;
```
default не обязательно

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+        }
+        switch (gender){
+            case MALE:
+                if(age>=30 && age<=55) rate=rate.subtract(manAgeRateDecrease);
```
тут можно чтобы не громоздить, сделать отдельный маленький метод и его вызывать

Skillshut replied:
и в одну строчку  лучше не писать условия с if

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+
+    }
+    public BigDecimal checkInsuranceEnabled(Boolean isInsuranceEnabled, BigDecimal rate) {
+        if(isInsuranceEnabled) rate=rate.subtract(BigDecimal.valueOf(3.0));
```
лучше в 2 строки и со скобками

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/ScoringService.java:
```diff
@@ -0,0 +1,144 @@
+        return rate;
+    }
+    public BigDecimal checkSalaryClient(Boolean isSalaryClient, BigDecimal rate) {
+        if (isSalaryClient) rate = rate.subtract(BigDecimal.valueOf(1.0));
```
лучше в 2 строки и со скобками

Skillshut commented on calculator/src/main/resources/META-INF/additional-spring-configuration-metadata.json:
```diff
@@ -0,0 +1,13 @@
+{
```
этот файл удалить

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/controller/LoanOfferControllerTest.java:
```diff
@@ -0,0 +1,59 @@
+    @BeforeEach
+    void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(loanOfferController).build();
+        objectMapper = new ObjectMapper();
```
зачем создавать objectMapper = new ObjectMapper(); для каждого теста? лучше его инициализировать 1 раз при объявлении

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/controller/LoanOfferControllerTest.java:
```diff
@@ -0,0 +1,59 @@
+    private MockMvc mockMvc;
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
```
и тут всего 1 тест, метод @BeforeEach вообще не имеет смысла

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/controller/ScoringControllerTest.java:
```diff
@@ -0,0 +1,61 @@
+    private ObjectMapper objectMapper;
+    @BeforeEach
+    void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(scoringController).build();
```
то же самое что и в LoanOfferControllerTest

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/controller/ScoringControllerTest.java:
```diff
@@ -0,0 +1,61 @@
+    }
+    @Test
+    public void scoringControllerTest() throws Exception {
+        ScoringDataDto sd = new ScoringDataDto(BigDecimal.valueOf(300000),6,"test","test",
```
используем builder lombok

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/controller/ScoringControllerTest.java:
```diff
@@ -0,0 +1,61 @@
+                "test",FEMALE, LocalDate.now().minusYears(40),"1234","123456",LocalDate.now(),"test",MARRIED,
+                0,new EmploymentDto(EMPLOYED,"123",BigDecimal.valueOf(100000),TOP_MANAGER,20,20),"12312",true,true);
+        String scoringJson = objectMapper.writeValueAsString(sd);
+        mockMvc.perform(post("/calculator/calc").contentType(MediaType.APPLICATION_JSON)
```
каждый вызов метода в цепочке с новой строки

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/service/LoanServiceTest.java:
```diff
@@ -0,0 +1,34 @@
+@ExtendWith(MockitoExtension.class)
+public class LoanServiceTest {
+    @InjectMocks
+    private LoanService loanService;
```
а что за моки инжектятся?

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/service/ScoringServiceTest.java:
```diff
@@ -0,0 +1,138 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ScoringServiceTest {
+    private ScoringService calculationCredit;
```
сервисы лучше через autowired в тестах, а не создавать руками

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/service/ScoringServiceTest.java:
```diff
@@ -0,0 +1,138 @@
+                "test",FEMALE, LocalDate.now().minusYears(30),"1234","123456",LocalDate.now(),"test",MARRIED,
+                0,new EmploymentDto(EMPLOYED,"123",BigDecimal.valueOf(100000),TOP_MANAGER,20,20),"12312",true,true);
+        BigDecimal actual = calculationCredit.makeScoring(scoringDataDTO, BigDecimal.valueOf(16.0));
+        System.out.println(actual);
```
лучше использовать логер ломбока также, если необходимо логировать

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/service/ScoringServiceTest.java:
```diff
@@ -0,0 +1,138 @@
+                "test",FEMALE, LocalDate.now().minusYears(40),"1234","123456",LocalDate.now(),"test",MARRIED,
+                0,new EmploymentDto(EMPLOYED,"123",BigDecimal.valueOf(100000),TOP_MANAGER,20,20),"12312",true,true);
+        BigDecimal actual = calculationCredit.makeScoring(scoringDataDTO, BigDecimal.valueOf(16.0));
+        System.out.println(actual);
```
везде поменять System.out.println(actual); на логгер

j-kachalova committed:
fix problems

changed logger, created services using Autowired, fixed code formatting, configured gitignore, fixed controllers

Skillshut commented on calculator/pom.xml:
```diff
@@ -0,0 +1,122 @@
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>properties-maven-plugin</artifactId>
+                <version>1.0-alpha-1</version>
```
а зачем этот плагин вообще нужен? и если нужен версию тоже вынести

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/LoanOfferController.java:
```diff
@@ -0,0 +1,33 @@
+@RestController
+public class LoanOfferController {
+    @Value("${app.keyRate}")
+    BigDecimal keyRate;
```
не поправила на private

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/LoanOfferController.java:
```diff
@@ -0,0 +1,33 @@
+
+@Slf4j
+@RestController
+public class LoanOfferController {
```
не сделала @ RequiredArgsConstructor вместо autowired

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/controller/ScoringController.java:
```diff
@@ -0,0 +1,48 @@
+
+@Slf4j
+@RestController
+public class ScoringController {
```
не сделала @ RequiredArgsConstructor вместо autowired

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/CreditDto.java:
```diff
@@ -0,0 +1,23 @@
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class CreditDto {
```
не добавила builder аннотацию, через нее будет создаваться класс

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/EmploymentDto.java:
```diff
@@ -0,0 +1,22 @@
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class EmploymentDto {
```
builder

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/LoanOfferDto.java:
```diff
@@ -0,0 +1,23 @@
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class LoanOfferDto {
```
builder

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/LoanStatementRequestDto.java:
```diff
@@ -0,0 +1,55 @@
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class LoanStatementRequestDto {
```
builder

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/PaymentScheduleElementDto.java:
```diff
@@ -0,0 +1,20 @@
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class PaymentScheduleElementDto {
```
builder

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/dto/ScoringDataDto.java:
```diff
@@ -0,0 +1,36 @@
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class ScoringDataDto {
```
builder

Skillshut commented on calculator/src/test/java/com/kachalova/calculator/service/CreditPaymentsServiceTest.java:
```diff
@@ -0,0 +1,80 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+//@RunWith(SpringRunner.class)
```
убрать, если не нужно, в коде должно быть в идеале 0% закомменитированного кода

Skillshut commented on calculator/src/main/java/com/kachalova/calculator/service/impl/ScoringServiceImpl.java:
```diff
@@ -0,0 +1,151 @@
+        log.info("rate after checkPosition is {}", userRate);
+        checkAmount(scoringDataDto.getAmount(), scoringDataDto.getEmployment().getSalary());
+        userRate = checkMaritalStatus(scoringDataDto.getMaritalStatus(), userRate);
+        log.info("rate after checkMaritalStatus is {}", userRate);
```
вот эти все промежуточные логи нужно на debug, а не info. на info уровне логируем только входные и выходные параметры метода, все остальное debug

j-kachalova committed:
refactor code
